packages/app/src/components/PageEditor/CommentMentionHelper.ts

```diff
@@ -0,0 +1,52 @@
+import { apiv3Get } from '~/client/util/apiv3-client';
+
+export default class CommentMentionHelper {
+
+  editor;
+
+  pattern: RegExp;
+
+
+  constructor(editor) {
+    this.editor = editor;
+    this.pattern = /@[A-Za-z0-9._-]{1,}/;
+  }
+
+  showUsernameHint = () => {
+    const currentPos = this.editor.getCursor();
+    const sc = this.editor.getSearchCursor(this.pattern, currentPos, { multiline:  false });
+    if (sc.findPrevious()) {
+      const isMentioning = (currentPos.line === sc.to().line && currentPos.ch === sc.to().ch);
+      if (!isMentioning) {
+        return;
+      }
+    }
+    else {
+      return;
+    }
```

Comment:
The cost of this implementation for searching `@username` pattern is too high.

- Just show hint only when the user input `@` and at least one letter instead of searching `@username` pattern from in the current line when each type by the user
- Use debounce if you need

Reply:
- Added condition to show hint only if length of mentioned username > 0
- Implement debounce when call showUsernameHint function

Reply:
I think this code take high cost because of searching.

`const sc = this.editor.getSearchCursor(this.pattern, currentPos, { multiline: false });`

How about the following code?

```

let mensionStartPos;
const isMenthoning = mensionStartPos != null;

if (!isMentioning) {
  const previousPos = { line: curPos.line, ch: curPos.ch - 1 };
  const previousChar = editor.getDoc().getRange(previousPos , curPos);

  if (previousChar !== '@') {
    return;  // I think the cost until this line is lower than searching
  }
  else {
    mendionStartPos = curPos;
  }
}
else {
  // get username that will be input after '@'
}
``

Reply:
I have made improvements to this code and removed the use of patern matching / regex